File: 0190-reverse-bits/0190-reverse-bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t result = 0;
        uint32_t rightest_bit;

        for (auto i = 0; i < 32; i++){
            rightest_bit = 1 & n;
            result = result << 1;
            result += rightest_bit;
            n = n >> 1;
        }

        return result;
    }
};

File: 0190-reverse-bits/0190-reverse-bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        reverseNum = 0

        for i in range(32):
            last_bit = n & 1
            n = n >> 1

            reverseNum = reverseNum << 1
            reverseNum |= last_bit

        return reverseNum
        